src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -40,6 +40,12 @@ protected override WorkspaceDiagnosticReport CreateReport(TextDocumentIdentifier
 
         protected override ImmutableArray<Document> GetOrderedDocuments(RequestContext context)
         {
+            // If we're being called from razor, we do not support WorkspaceDiagnostics at all.  For razor, workspace
+            // diagnostics will be handled by razor itself, which will operate by calling into Roslyn and asking for
+            // document-diagnostics instead.
+            if (context.ClientName != null)
```

Comment:
At this point, I feel like we have enough checks for this sort of thing that having a constant for the actual Razor client name might just be easier :)

Reply:
agreed.  i think this all just boils down to a .IsRazor bit.